api/hypershift/v1alpha1/nodepool_types.go

```diff
@@ -888,16 +888,55 @@ type AgentNodePoolPlatform struct {
 	AgentLabelSelector *metav1.LabelSelector `json:"agentLabelSelector,omitempty"`
 }
 
+type AzureVMImageType string
```

Comment:
why are we adding stuff into alpha API?

Reply:
If its not there, the unit tests are failing on `TestFuzzyConversion/for_NodePool`